contracts/scripts/deploy.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @notice Umbra Deployment script
+ * @dev At each contract deployment or transaction, we wait for the transaction to be mined before
+ * continuing to the next step. This is done because it makes it simpler to continue from that spot
+ * rather than restart the full deployment
+ * @dev To initialize a deploy:
+ *   `yarn deploy:umbra --network <network>`   (where network specifies a netwrk found in the hardhat.config.ts file)
+ * If deploying to a local node (--network localhost), first start Hardhat using `yarn hardhat node`
+ */
+const fs = require('fs');
+const hre = require('hardhat');
+const { ethers } = hre;
+
+const RINKEBY_PAYMASTER_PUBLIC_RELAYER_ADDRESS = '0x53C88539C65E0350408a2294C4A85eB3d8ce8789';
+
+const deployParams = require('./deployParams.json');
+
+// Initialize object that will hold all deploy info. We'll continually update this and save it to
+// a file using the save() method below
+const parameters = {
+  // TODO consider changing layout so a contract's constructor parameters live with its address
+  admin: null,
+  contracts: {}, // will be populated with all contract addresses
+  actions: {}, // will be populated with deployment actins
+};
+
+// Setup for saving off deploy info to JSON file
+const now = new Date().toISOString();
+const folderName = './deploy-history';
+const fileName = `${folderName}/${now}.json`;
+fs.mkdir(folderName, (err) => {
+  if (err && err.code !== 'EEXIST') throw err;
+});
+
+// method to save the deploy info to a JSON file
+const save = (value, field, subfield = undefined) => {
+  if (subfield) {
+    parameters[field][subfield] = value;
+  } else {
+    parameters[field] = value;
+  }
+  fs.writeFileSync(fileName, JSON.stringify(parameters));
```

Comment:
We should modify the output file name to include the network name, then modify the `.gitignore` to exclude all localhost deploy files but commit the rest. That way the most recent deploy on any network always lives in the repo

Reply:
Another good idea.. done